### PR TITLE
Ignored mobs (hotfix + color stripping support)

### PR DESCRIPTION
**Pull request by mrgeneralq**
- Fixed the bug where it would always ignore nameless mobs
- Added color, config.yml no longer requires the color tags in order to detect the mob.

**IMPORTANT**: the ignored-mobs section will override the capture-named-mobs property

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java [hunk 1]
```diff
@@ -75,14 +77,20 @@ protected boolean canCapture(Player p, LivingEntity entity) {
         }
 
         List<String> ignoredMobNames = MobCapturer.getRegistry().getConfig().getStringList("options.ignored-mobs");
-        if (ignoredMobNames.contains(entity.getCustomName())) {
-            // filter out ignored mob names
-            return false;
+        if (ignoredMobNames.size() > 0){
+            String strippedEntityName = ChatColor.stripColor(entity.getCustomName());
+            List<String> strippedMatches = ignoredMobNames.stream()
```
Comment (ybw0014) [inline src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]: The list is usually not big, so I don't think we need stream here, a for loop should be enough

**Review by ybw0014 — CHANGES_REQUESTED** [review]
Also I think we should check `capture-named-mobs` first, if this option is disabled, we don't need to run ignored names check.

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java (hunk 1 above)
Comment (mrgeneralq) [reply src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]: - Replaced stream with for loop
- Moved ignored-mobs functionality below the capture-named-mobs

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java [hunk 2]
```diff
@@ -74,15 +76,20 @@ protected boolean canCapture(Player p, LivingEntity entity) {
...
         }
 
-        if (MobCapturer.getRegistry().getConfig().getBoolean("options.capture-named-mobs")) {
-            // check if the mob has a name, if not return false
-            return entity.getCustomName() != null;
+        List<String> ignoredMobNames = MobCapturer.getRegistry().getConfig().getStringList("options.ignored-mobs");
+        if (ignoredMobNames.size() > 0){
+            String strippedEntityName = ChatColor.stripColor(entity.getCustomName());
+            for(String ignoredMobName: ignoredMobNames){
+                if(ignoredMobName.equalsIgnoreCase(strippedEntityName)){
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]:
```suggestion
                if (ignoredMobName.equalsIgnoreCase(strippedEntityName)) {
```

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java [hunk 3]
```diff
@@ -74,15 +76,20 @@ protected boolean canCapture(Player p, LivingEntity entity) {
...
+            }
         }
 
-        if (MobCapturer.getRegistry().getConfig().getBoolean("options.capture-named-mobs")) {
-            // check if the mob has a name, if not return false
-            return entity.getCustomName() != null;
+        List<String> ignoredMobNames = MobCapturer.getRegistry().getConfig().getStringList("options.ignored-mobs");
+        if (ignoredMobNames.size() > 0){
+            String strippedEntityName = ChatColor.stripColor(entity.getCustomName());
+            for(String ignoredMobName: ignoredMobNames){
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]:
```suggestion
            for (String ignoredMobName : ignoredMobNames) {
```

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java [hunk 4]
```diff
@@ -74,15 +76,20 @@ protected boolean canCapture(Player p, LivingEntity entity) {
...
 
-        List<String> ignoredMobNames = MobCapturer.getRegistry().getConfig().getStringList("options.ignored-mobs");
-        if (ignoredMobNames.contains(entity.getCustomName())) {
-            // filter out ignored mob names
-            return false;
+        if (!MobCapturer.getRegistry().getConfig().getBoolean("options.capture-named-mobs")) {
+            if (entity.getCustomName() != null){
+                return false;
+            }
         }
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]:
```suggestion
        if (!MobCapturer.getRegistry().getConfig().getBoolean("options.capture-named-mobs") 
        && entity.getCustomName() != null) { 
        	return false;
        }
```

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java [hunk 5]
```diff
@@ -2,10 +2,12 @@
 
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
```
Comment (ybw0014) [inline src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]: is this used anywhere?
```suggestion
```

File: src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java [hunk 6]
```diff
@@ -74,15 +76,19 @@ protected boolean canCapture(Player p, LivingEntity entity) {
             return false;
         }
 
-        List<String> ignoredMobNames = MobCapturer.getRegistry().getConfig().getStringList("options.ignored-mobs");
-        if (ignoredMobNames.contains(entity.getCustomName())) {
-            // filter out ignored mob names
-            return false;
+        if (!MobCapturer.getRegistry().getConfig().getBoolean("options.capture-named-mobs") 
+        && entity.getCustomName() != null) {
```
Comment (ybw0014) [inline src/main/java/io/github/thebusybiscuit/mobcapturer/listeners/PelletListener.java]:
```suggestion
            && entity.getCustomName() != null) { 
```

**Review by ybw0014 — CHANGES_REQUESTED** [review]

**Review by ybw0014 — APPROVED** [review]

**Pull request merged**